openssl/src/ssl/mod.rs

```diff
@@ -312,6 +321,22 @@ impl SslContext {
             })
     }
 
+    /// Specifies the private key
+    pub fn set_private_key(&mut self, key: &PKey) -> Option<SslError> {
+        wrap_ssl_result(
+            unsafe {
+                ffi::SSL_CTX_use_PrivateKey(*self.ctx, key.get_handle())
+            })
+    }
+
+    /// Check consistency of private key and certificate
+    pub fn check_private_key(&mut self) -> Option<SslError> {
```

Comment:
It's a bit nicer to use `Result<(), SslError>` rather than `Option<SslError>` so that things compose more nicely with `try!` etc.